File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> num_to_index;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (num_to_index.count(complement)) {
                return {i, num_to_index[complement]};
            }
            num_to_index[nums[i]] = i;
        }

        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_index = {}
        
        for i, num in enumerate(nums):
            if (complement := target - num) in num_to_index:
                return [i, num_to_index[complement]]
            num_to_index[num] = i

        return []